src/screens/Customers/CustomerMain.tsx

```diff
@@ -1,29 +1,130 @@
 import React, { useEffect, useState } from 'react';
-import { CustomerRecord } from '../../lib/airtable/interface';
-import { getAllCustomers } from '../../lib/airtable/request';
+import { CustomerRecord, SiteRecord } from '../../lib/airtable/interface';
 import { Link, RouteComponentProps } from 'react-router-dom';
+import { store } from '../../lib/redux/store';
+import { createStyles, FormControl, FormHelperText, MenuItem, Select, Theme, withStyles } from '@material-ui/core';
+import UserSearchBar from '../../components/UserSearchBar';
+import CustomerCard from '../../components/CustomerCard';
+import TrieTree from '../../lib/utils/TrieTree';
 
-function CustomerMain(props: RouteComponentProps) {
+const styles = (theme: Theme) =>
+  createStyles({
+    title: {
+      width: '153.9px',
+      height: '33.83px',
+      left: '25.6px',
+      top: '96px',
+      fontFamily: 'Helvetica Neue',
+      fontStyle: 'normal',
+      fontWeight: 500,
+      fontSize: '30px',
+      lineHeight: '115%',
+      color: '#828282',
+      flexGrow: 1,
+    },
+    headerWrapper: {
+      display: 'flex',
+      flexDirection: 'row',
+      width: '100%',
+      paddingTop: '86px',
+      paddingRight: '21px',
+      paddingLeft: '21px',
+    },
+    selectionHeader: {
+      display: 'flex',
+      flexDirection: 'column',
+      flexGrow: 1,
+    },
+    scrollDiv: {
+      maxHeight: '380px',
+      overflow: 'auto',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      width: '310.71px',
+    },
+  });
+
+interface UserProps {
+  classes: { title: string; headerWrapper: string; selectionHeader: string; scrollDiv: string };
+}
+
+function CustomerMain(props: RouteComponentProps & UserProps) {
   useEffect(() => {
     getCustomers();
   }, []);
```

Comment:
Performance might suffer here since we call `getCustomers` (which iterates through all customers and does some sets) every time we do a rerender, which happens when we search. We might just want to call this `getCustomers` on the initial render to save all the extra work when we render during filters. Open to discuss on this though if you have opinions